### PR TITLE
[Dropdown] Updated stylings, states and stories

### LINKED ISSUE #173: [Component] Dropdown - Updated styling, optimized state usage
## Changes
- [x] Uses the standard CFPB accessibility styling (dotted outline when focused) in the Options 
- [x] Incorporate an `error` prop ---> border turns red
- [x] Utilize the current library's Checkbox instead of a generic checkbox
- [x] Apply "single source of truth" (removed internal `useState`) to both Checkbox and Dropdown. State must come externally in a `controlled` manner.
- [x] Update TypeScript settings
- [x] Update the Multi-Select variation
- [x] Incorporate the Menu Options styling
- [x] Incorporate the styling from Figma
- [x] Update the storybook stories of Checkbox and Dropdown to use external state.
---
## References
- https://github.com/cfpb/sbl-frontend/issues/41
- https://cfpb.github.io/design-system/components/dropdowns-and-multiselects

### PR DESCRIPTION
**Pull request by shindigira**
Closes #173 

## Changes

- [New] Uses the standard CFPB accessibility styling (dotted outline when focused) in the Options 
- [New] Utilizes the proper  CFPB-style 'down' [&#9660;](http://www.fileformat.info/info/unicode/char/25bc/index.htm) icon in the Dropdown
- [New] `error` prop that causes a red border around the Dropdown
- [New] Dropdown Storybook story ('With Error')
- [Update] Utilize the current library's Checkbox instead of a generic checkbox
- [Update] Removed internal useState usage in **_both_** Checkbox and Dropdown. State must come externally in a `controlled` manner.
- [Update] Update the storybook stories of Checkbox and Dropdown to use external state.
- [Update] Update TypeScript settings
- [Update] Update the Multi-Select variation
- [Update] Incorporate the Menu Options styling
- [Update] Incorporate the styling from Figma

## How to test this PR

1. `yarn vitest`
2. See that everything passes
3. `yarn storybook`
4. Test the Checkbox and Dropdown stories
5. Inspect the new styling and compare with the figma and CFPB design

### Default
Screenshot 2023-09-19 at 4 01 28 PM

### Error
Screenshot 2023-09-19 at 4 01 09 PM

### Multiselect with Pills and Checkboxes
Screenshot 2023-09-19 at 4 02 01 PM

### Multiselect -- Menu Options (Dotted Outline While Focused)
Screenshot 2023-09-25 at 9 53 40 AM

### CODE REVIEW

**Review by meissadia — CHANGES_REQUESTED** [review]
- Should we try to better align stories to what is in the DS?
  - Sigmund has done a great job of demonstrating what is possible with the Dropdown component but a lot of the scenarios are not represented in the DS ("With error", "Disabled", "Multi-select with checkboxes", etc).  I could use some clarification on the source for all of the features that are not outlined in the DS.

**Review by meissadia — COMMENTED** [review]
- I pushed a commit to update story names to `sentence case`
- I pushed a commit to update `Dropdown` to `Dropdowns`, to align with the way we've been updating the naming of Story pages

**Comment by meissadia** [comment]
I had also previously added a `Clear All` button to the selected pills section and I'm struggling to find a reference for the source (Not seeing it in the DS or in the Figma).  Should this be removed?

Screenshot 2023-09-19 at 3 50 17 PM

File: src/components/Dropdown/Dropdown.less [hunk 1]
```diff
@@ -0,0 +1,52 @@
+.dropdown-control {
+  border-radius: 0 !important;
+  border-width: 1px !important;
+  width: 100% !important;
+  border-color: #919395 !important;
+  // outline: solid #919395 !important;
```
Comment (meissadia) [inline src/components/Dropdown/Dropdown.less:6]:
```suggestion
```

File: src/components/Dropdown/Dropdown.less [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  border-color: inherit !important;
+}
+
+.dropdown-value-container {
+}
+
+.dropdown-value-container--error {
+  border-style: solid !important;
+  border-color: #d14124 !important;
+  // outline: solid #d14124 !important;
```
Comment (meissadia) [inline src/components/Dropdown/Dropdown.less:42]:
```suggestion
```

File: src/components/Checkbox/Checkbox.stories.tsx [hunk 3]
```diff
@@ -1,23 +1,64 @@
...
+        })
+      }
+    />
+  );
+}
+
 export const DefaultCheckbox: Story = {
-  args: { id: 'default', name: 'default', label: 'Default checkbox' }
+  render: _args => CheckboxWrapper(_args),
+  name: 'Default dropdown',
```
Comment (meissadia) [inline src/components/Checkbox/Checkbox.stories.tsx:51]:
```suggestion
  name: 'Default checkbox',
```

File: src/components/Dropdown/Dropdown.less [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  border-left: #919395 1px solid;
+}
+
+.dropdown-dropdown-indicator {
+  color: inherit !important;
+  border-color: inherit !important;
+}
+
+.dropdown-value-container {
+}
```
Comment (meissadia) [inline src/components/Dropdown/Dropdown.less:37]: Empty selector needed?
```suggestion
```

File: src/components/Dropdown/Dropdown.less [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+.dropdown-value-container--error {
+  border-style: solid !important;
+  border-color: #d14124 !important;
+  // outline: solid #d14124 !important;
+  border-width: 0.17em !important;
+}
+
+.dropdown-value-container--success {
+}
```
Comment (meissadia) [inline src/components/Dropdown/Dropdown.less:47]: Empty selector needed?
```suggestion
```

File: src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -110,19 +146,20 @@ export function Dropdown({
...
         onKeyDown={onKeyDown}
         openMenuOnFocus
-        options={filterOptions(options, selected, isMulti, withCheckbox)}
+        options={filterOptions(options, value, Boolean(isMulti), withCheckbox)}
         ref={selectReference as Ref<any>}
         styles={extendedSelectStyles}
         tabSelectsValue={false}
-        value={value ?? selected}
+        value={value}
+        // menuIsOpen
```
Comment (meissadia) [inline src/components/Dropdown/Dropdown.tsx:154]: Remove unused prop? 
```suggestion
```

File: src/components/Dropdown/DropdownInputWithCheckbox.tsx [hunk 7]
```diff
@@ -26,15 +27,24 @@ const CheckboxInputOption = ({
...
 
   // styles
-  let bg = 'transparent';
-  if (isFocused) bg = '#eee';
-  if (isActive) bg = '#B2D4FF';
+  const isFocusedStyles = {
+    backgroundColor: '#B2D4FF'
+    // backgroundColor: '#eee',
+    // outline: '1px dotted #0072ce'
+    // outlineOffset: '3px'
```
Comment (meissadia) [inline src/components/Dropdown/DropdownInputWithCheckbox.tsx:34]:
```suggestion
```

File: src/components/Dropdown/DropdownInputWithCheckbox.tsx [hunk 8]
```diff
@@ -26,15 +27,24 @@ const CheckboxInputOption = ({
...
+  };
+
+  const isActiveStyle = {
+    // backgroundColor: '#B2D4FF'
+  };
```
Comment (meissadia) [inline src/components/Dropdown/DropdownInputWithCheckbox.tsx:39]: Remove style object that is not currently applying any styles?
```suggestion
```

File: src/components/Dropdown/DropdownInputWithCheckbox.tsx [hunk 9]
```diff
@@ -46,23 +56,25 @@ const CheckboxInputOption = ({
     style
   };
 
+  console.log('rest:', rest);
```
Comment (meissadia) [inline src/components/Dropdown/DropdownInputWithCheckbox.tsx:59]:
```suggestion
```

File: src/components/Dropdown/DropdownInputWithCheckbox.tsx [hunk 10]
```diff
@@ -46,23 +56,25 @@ const CheckboxInputOption = ({
...
+
   return (
     <components.Option
-      className='OptionWithCheckbox'
-      getStyles={getStyles}
+      className='option-with-checkbox'
+      // getStyles={getStyles}
```
Comment (meissadia) [inline src/components/Dropdown/DropdownInputWithCheckbox.tsx:64]:
```suggestion
```

File: src/components/Dropdown/styles.tsx [hunk 11]
```diff
@@ -18,5 +14,12 @@ export const extendedSelectStyles = {
         : '1px solid #0072ce !important',
       outlineOffset: state.isFocused ? '3px' : '0'
     }
+  }),
+  menu: provided => ({
+    ...provided,
+    // backgroundColor: '#f0f0f0', // Background color of the entire menu
```
Comment (meissadia) [inline src/components/Dropdown/styles.tsx:20]:
```suggestion
```

File: src/components/Checkbox/Checkbox.stories.tsx [hunk 12]
```diff
@@ -1,23 +1,64 @@
+/* eslint-disable unicorn/prevent-abbreviations */
```
Comment (meissadia) [inline src/components/Checkbox/Checkbox.stories.tsx:1]: Necessary? I'm not sure which plugin it is but my VS Code automatically updates names such as `_args` for me upon save.  For consistency, I think this should stay enabled. 
```suggestion
```

**Review by meissadia — CHANGES_REQUESTED** [review]
Functionality looks great! Could use some code cleanup.
 
- Border for Dropdown -> Error feels a bit heavy.  I think I've seen it mentioned that it should be 2px?
- Still in discussion but we should match the DS multiselect's :focus and :hover styles as closely as possible for now

**Comment by natalia-fitzgerald** [comment]
@shindigira 
- For the Multiselect remove the "x" within the input field (and the gray box)
  - We can handle the clear function at the form level
- For the Multiselect use the standard CFPB styling for the focus state
  - Changes to the focus state would be global and affect the rest of our CFPB custom components (this is out of scope for this effort)

There are known bugs in the CFPB Design System - Multiselect component that we should keep in mind as we build out this component. We can then contribute back to the system while we build the React component library. 

**Some known bugs are documented here:** 

https://github.com/cfpb/design-system/issues?q=is%3Aissue+is%3Aopen+multiselect
https://github.cfpb.gov/CFGOV/platform/issues/3291
https://github.cfpb.gov/CFGOV/platform/issues/3283

**Review by natalia-fitzgerald — COMMENTED** [review]
@shindigira 
- For the Multiselect remove the "x" within the input field (and the gray box)
  - We can handle the clear function at the form level
- For the Multiselect use the standard CFPB styling for the focus state
  - Changes to the focus state would be global and affect the rest of our CFPB custom components (this is out of scope for this effort)

There are known bugs in the CFPB Design System - Multiselect component that we should keep in mind as we build out this component. If we have ideas about how to solve these issues we can contribute back to the system while we build the React component library. 

**Some known bugs are documented here:** 

https://github.com/cfpb/design-system/issues?q=is%3Aissue+is%3Aopen+multiselect
https://github.cfpb.gov/CFGOV/platform/issues/3291
https://github.cfpb.gov/CFGOV/platform/issues/3283
https://github.com/cfpb/design-system/issues/1728

File: src/components/Dropdown/Dropdown.less [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  border-color: #0072ce !important;
+  outline: 1px dotted #0072ce !important;
+  outline-offset: 2px !important;
+}
+
+// .dropdown-control:hover {
+//   border-color: #0072ce !important;
+//   outline: 1px dotted #0072ce !important;
+//   outline-offset: 2px !important;
+// }
```
Comment (meissadia) [inline src/components/Dropdown/Dropdown.less:18]:
```suggestion
```

File: src/components/Dropdown/Dropdown.tsx [hunk 14]
```diff
@@ -1,24 +1,51 @@
...
+  </components.DropdownIndicator>
+);
+const customClearIndicator = (properties): JSX.Element => (
+  <components.ClearIndicator {...properties}>
+    <div className='dropdown-clear-indicator-container'>
+      <Icon name='error' />
+    </div>
+  </components.ClearIndicator>
+);
+
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/components/Dropdown/Dropdown.tsx]: Per a convo w/ Natalia, I've pushed a change to clear the background styling

When Dropdown not focused
Screenshot 2023-09-26 at 5 29 41 PM
 
When Dropdown is focused (x is darker)
Screenshot 2023-09-26 at 5 29 49 PM

**Comment by meissadia** [comment]
I pushed some changes to:
- Fix the Dropdown interactions on the Overview page
- Remove custom styles on the in-field `Clear all` "X"
  - Screenshot 2023-09-26 at 5 24 46 PM
- Address some grey border overhang on the Dropdown error state
  - Before / After
  - Screenshot 2023-09-26 at 5 37 04 PM

**Review by meissadia — APPROVED** [review]
@shindigira Please take a look at the adjustments I made to confirm you're okay with them

**Pull request merged**